File: Frontend/MovieBookingAppUi/src/app/services/ticket.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Ticket } from '../ticket/ticket';
import { TicketDto } from '../ticket/ticket-dto';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  data: any;

  constructor(private http: HttpClient) { }

  BASIC_URL = 'http://localhost:8080/api/v1.0/moviebooking';
  
  public saveTicket(id: number, ticket: TicketDto) {
    return this.http.post(this.BASIC_URL + "/ticket/" + id, ticket);
  }

  public myTickets(userName: string) {
    return this.http.get<Ticket[]>(this.BASIC_URL + "/ticket/" + userName);
  }

  public allTickets() {
    return this.http.get<Ticket[]>(this.BASIC_URL + "/tickets");
  }
}

File: Backend/Movie/src/main/java/com/cts/movie/services/TicketService.java
package com.cts.movie.services;

import java.util.List;

import com.cts.movie.dto.TicketDTO;
import com.cts.movie.model.Ticket;

public interface TicketService {

	public List<Ticket> fetchAllTicketByUserName(String userName);

	public List<Ticket> fetchAllTickets();

	public Ticket saveTicket(int id, TicketDTO ticketDto);

}
